### ISSUE TITLE
Compilation error when 2 Oracle packages both contain the same record type name

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10576]
The following setup produces a compilation error as the generated record type reference is duplicated in the `UDT.java` class:

```sql
CREATE OR REPLACE PACKAGE p AS
  TYPE t IS RECORD (a NUMBER);
END p;
/
  
CREATE OR REPLACE PACKAGE p AS
  TYPE t IS RECORD (a NUMBER);
END p;
/
```

**Comment by lukaseder** [comment]
This only affects the global references in the `UDTs.java` class, not the types themselves, nor their usage.

Workarounds include:

- Turning off `<globalUDTReferences>false</globalUDTReferences>`
- Using a generator strategy
- Excluding the relevant packages or types using `<excludes/>`

**Comment by lukaseder** [comment]
The problem is that these `UDT` types are generated in the wrong place, in the schema's `UDTs.java` file, when there should also be such a file per package. Changing this is backwards incompatible, as user code will no longer compile if someone references the `UDTs` this way. But it seems necessary, because the package UDTs can also collide with `OBJECT` types, this way.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.14.0.

The fix depends on new API and breaks compatibility of generated code. I prefer not to backport it to 3.13.5. One of the existing workarounds documented here (https://github.com/jOOQ/jOOQ/issues/10576#issuecomment-685720508) should be used.
